### PR TITLE
Add filetype configuration menu entries

### PR DESCRIPTION
**Pull request by techee**
This is the separated part from https://github.com/geany/geany/pull/485 adding just the menu entries. All issues from the previous patch should be addressed here.

### CODE REVIEW

File: src/filetypes.c [hunk 1]
```diff
@@ -402,29 +419,39 @@ static void create_sub_menu(GtkWidget *parent, GeanyFiletypeGroupID group_id, co
...
+		GtkWidget *parent = (ft->group != GEANY_FILETYPE_GROUP_NONE) ? group_menus[ft->group] : menu;
+
+		/* we already have filetypes.common config entry */
+		if (config && ft->id == GEANY_FILETYPES_NONE)
+			continue;
 
-		if (ft->group != GEANY_FILETYPE_GROUP_NONE)
-			create_radio_menu_item(group_menus[ft->group], ft);
+		if (config)
+			ui_add_config_file_menu_item(filetypes_get_filename(ft, TRUE), NULL, GTK_CONTAINER(parent));
```
Comment (b4n) [inline src/filetypes.c]: `filetypes_get_filename()` return value is leaked

File: src/filetypes.c [hunk 2]
```diff
@@ -354,28 +354,43 @@ void filetypes_init_types(void)
...
-		guint i;
+		for (i = 0; i < filetypes_array->len; i++)
+		{
+			GeanyFiletype *ft = filetypes[i];
 
-		/* Note: we don't reload other filetypes, even though the named styles may have changed.
-		 * The user can do this manually with 'Tools->Reload Configuration' */
-		filetypes_load_config(GEANY_FILETYPES_NONE, TRUE);
+			f = g_strconcat("filetypes.", filetypes_get_conf_extension(ft), NULL);
+			SETPTR(f, g_build_filename(app->configdir, GEANY_FILEDEFS_SUBDIR, f, NULL));
```
Comment (b4n) [inline src/filetypes.c]: AFAICT you could use filetypes_get_filename() instead of duplicating the logic

Comment (b4n) [inline src/filetypes.c]: especially as `filetypes_get_conf_extension()` retuns an allocated string, so there's a leak :)

**Comment by b4n** [comment]
apart that LGTM and seems to work great :)

File: src/filetypes.c [hunk 3]
```diff
@@ -354,28 +354,43 @@ void filetypes_init_types(void)
 
 static void on_document_save(G_GNUC_UNUSED GObject *object, GeanyDocument *doc)
 {
-	gchar *f;
+	gchar *f, *basename;
+	gint i;
```
Comment (b4n) [inline src/filetypes.c]: should be unsigned, and AFAICT could go back in the deeper scope

**Comment by b4n** [comment]
Suggested changes:
```diff
diff --git a/src/filetypes.c b/src/filetypes.c
index f9c21c0..6651a04 100644
--- a/src/filetypes.c
+++ b/src/filetypes.c
@@ -355,7 +355,6 @@ void filetypes_init_types(void)
 static void on_document_save(G_GNUC_UNUSED GObject *object, GeanyDocument *doc)
 {
 	gchar *f, *basename;
-	gint i;
 
 	g_return_if_fail(!EMPTY(doc->real_path));
 
@@ -367,12 +366,13 @@ static void on_document_save(G_GNUC_UNUSED GObject *object, GeanyDocument *doc)
 	basename = g_path_get_basename(doc->real_path);
 	if (g_str_has_prefix(basename, "filetypes."))
 	{
+		guint i;
+
 		for (i = 0; i < filetypes_array->len; i++)
 		{
 			GeanyFiletype *ft = filetypes[i];
 
-			f = g_strconcat("filetypes.", filetypes_get_conf_extension(ft), NULL);
-			SETPTR(f, g_build_filename(app->configdir, GEANY_FILEDEFS_SUBDIR, f, NULL));
+			f = filetypes_get_filename(ft, TRUE);
 			if (utils_str_equal(doc->real_path, f))
 			{
 				guint j;
@@ -449,7 +449,12 @@ static void create_set_filetype_menu(gboolean config)
 			continue;
 
 		if (config)
-			ui_add_config_file_menu_item(filetypes_get_filename(ft, TRUE), NULL, GTK_CONTAINER(parent));
+		{
+			gchar *filename = filetypes_get_filename(ft, TRUE);
+
+			ui_add_config_file_menu_item(filename, NULL, GTK_CONTAINER(parent));
+			g_free(filename);
+		}
 		else
 			create_radio_menu_item(parent, ft);
 	}
```

**Comment by techee** [comment]
@b4n Thanks for being my leak-detector, apparently I need to use tools like you :-).

I've just repushed with the suggested changes.

**Pull request merged**